src/oc/storage/api/activity.clj

```diff
@@ -0,0 +1,157 @@
+                             user-id (:user-id user)
+                             org (:existing-org ctx)
+                             org-id (:uuid org)
+                             ctx-params (keywordize-keys (-> ctx :request :params))
+                             start? (if (:start ctx-params) true false) ; flag if a start was specified
+                             start-params (update ctx-params :start #(or % (db-common/current-timestamp))) ; default is now
+                             direction (or (#{:after :around} (keyword (:direction ctx-params))) :before) ; default is before
+                             ;; around is only valid with a specified start
+                             allowed-direction (if (and (not start?) (= direction :around)) :before direction) 
+                             params (merge start-params {:direction allowed-direction :start? start?})
+                             boards (board-res/list-boards-by-org conn org-id [:created-at :updated-at :authors :viewers :access])
+                             ;allowed-boards (map :uuid (filter #(access/access-level-for org % user) boards))
+                             board-uuids (map :uuid boards)
+                             board-slugs-and-names (map #(array-map :slug (:slug %) :name (:name %)) boards)
+                             board-by-uuid (zipmap board-uuids board-slugs-and-names)
+                             activity (assemble-activity conn params org board-by-uuid)]
+                          (activity-rep/render-activity-list params org activity (:access-level ctx) user-id))))
+
+;; A resource for operations on the calendar of activity for a particular Org
+(defresource calendar [conn slug]
+  (api-common/open-company-anonymous-resource config/passphrase) ; verify validity of optional JWToken
+
+  :allowed-methods [:options :get]
+
+  ;; Media type client accepts
+  :available-media-types [mt/activity-calendar-media-type]
+  :handle-not-acceptable (api-common/only-accept 406 mt/activity-calendar-media-type)
+
+  ;; Authorization
+  :allowed? (by-method {
+    :options true
+    :get (fn [ctx] (access/access-level-for conn slug (:user ctx)))})
+
+  ;; Existentialism
+  :exists? (fn [ctx] (if-let* [_slug? (slugify/valid-slug? slug)
+                               org (or (:existing-org ctx) (org-res/get-org conn slug))]
+                        {:existing-org org}
+                        false))
+
+  ;; Responses
+  :handle-ok (fn [ctx] (let [user (:user ctx)
+                             user-id (:user-id user)
+                             org (:existing-org ctx)
+                             org-id (:uuid org)
+                             ;boards (board-res/list-boards-by-org conn org-id [:created-at :updated-at :authors :viewers :access])
+                             ;allowed-boards (map :uuid (filter #(access/access-level-for org % user) boards))
+                             ;board-uuids (map :uuid boards)
```

Comment:
@bago2k4 as a reminder, I need to filter out private boards from the user.

The reminder did its job!